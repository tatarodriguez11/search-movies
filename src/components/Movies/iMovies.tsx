import { Film } from "../../interfaces";

export interface ListOfMoviesProps {
  movies: Film[]
}