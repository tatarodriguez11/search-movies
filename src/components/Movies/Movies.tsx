import { ListOfMoviesProps } from "./iMovies";
import ListOfMovies from "../ListOfMovies/ListOfMovies";
import { NoMoviesResults } from "../NoMoviesResult/NoMoviesResult";
import './Movies.scss'
export function Movies({ movies }: ListOfMoviesProps) {
  const hasMovies = movies?.length > 0;

  return hasMovies ? <ListOfMovies movies={movies} /> : <NoMoviesResults />;
}
