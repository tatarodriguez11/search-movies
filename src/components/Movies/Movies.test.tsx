import React from "react";
import { getByRole, render } from "@testing-library/react";
import { Movies } from "./Movies";
import { BrowserRouter } from "react-router-dom";

describe("Movies component", () => {
  const movies = [
    {
      id: "1",
      title: "The Godfather",
      image: "https://example.com/movie.jpg",
      year: "1972",
    },
    {
      id: "2",
      title: "Pulp Fiction",
      image: "https://example.com/movie.jpg",
      year: "1994",
    },
    {
      id: "3",
      title: "The Shawshank Redemption",
      image: "https://example.com/movie.jpg",
      year: "1994",
    },
  ];

  it("should match the snapshot when there are movies", () => {
    const { container } = render(
      <BrowserRouter>
        <Movies movies={movies} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it("should match the snapshot when there are no movies", () => {
    const { container } = render(<Movies movies={[]} />);
    expect(container).toMatchSnapshot();
  });

  it("should render ListOfMovies component when there are movies", () => {
    const { getByRole } = render(
      <BrowserRouter>
        <Movies movies={movies} />
      </BrowserRouter>
    );
    expect(getByRole("list")).toBeInTheDocument();
  });

  it("should render NoMoviesResults component when there are no movies", () => {
    const { getByText } = render(<Movies movies={[]} />);
    expect(
      getByText("No se encontraron películas para esta búsqueda")
    ).toBeInTheDocument();
  });
});
