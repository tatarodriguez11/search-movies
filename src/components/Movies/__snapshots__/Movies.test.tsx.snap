// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies component should match the snapshot when there are movies 1`] = `
<div>
  <ul
    aria-role="list"
    class="movies"
  >
    <li
      class="movie"
    >
      <a
        href="/movies/1"
      >
        <h3
          class="movie__title"
        >
          The Godfather
        </h3>
        <p
          class="movie__year"
        >
          1972
        </p>
        <img
          alt="The Godfather"
          class="movie__image"
          src="https://example.com/movie.jpg"
        />
      </a>
    </li>
    <li
      class="movie"
    >
      <a
        href="/movies/2"
      >
        <h3
          class="movie__title"
        >
          Pulp Fiction
        </h3>
        <p
          class="movie__year"
        >
          1994
        </p>
        <img
          alt="Pulp Fiction"
          class="movie__image"
          src="https://example.com/movie.jpg"
        />
      </a>
    </li>
    <li
      class="movie"
    >
      <a
        href="/movies/3"
      >
        <h3
          class="movie__title"
        >
          The Shawshank Redemption
        </h3>
        <p
          class="movie__year"
        >
          1994
        </p>
        <img
          alt="The Shawshank Redemption"
          class="movie__image"
          src="https://example.com/movie.jpg"
        />
      </a>
    </li>
  </ul>
</div>
`;

exports[`Movies component should match the snapshot when there are no movies 1`] = `
<div>
  <p>
    No se encontraron películas para esta búsqueda
  </p>
</div>
`;
